src/server/system_services/pool_controllers.js

```diff
@@ -271,84 +231,44 @@ class InProcessAgentsPoolController extends PoolController {
     }
 }
 
-// ----------------------------------------------
-// Shared utils
-// ---------------------------------------------
-function _get_statefulset_name(system_name, pool_name) {
-    return `${pool_name}-${system_name}-noobaa`;
-}
-
 async function _get_k8s_conf(params) {
     const yaml_path = path.resolve(__dirname, '../../deploy/NVA_build/noobaa_pool.yaml');
-    const yaml = (await fs.readFileAsync(yaml_path)).toString();
-    const conf_template = await yaml_utils.parse(yaml, true);
-
-     // Find the noobaa-agent stateful set.
-    const statefulset = conf_template.items
-        .find(resource =>
-            resource.kind === 'StatefulSet' &&
-            resource.metadata.name === 'noobaa-agent'
-        );
-    if (!statefulset) {
-        throw new Error('Invalid agent template: missing StatefulSet named noobaa-agent');
-    }
-
-    // A ref to the template section.
-    const { template } = statefulset.spec;
-
-    // Find the noobaa agent container.
-    const agent_container = template.spec.containers
-        .find(container => container.name === 'noobaa-agent');
-    if (!agent_container) {
-        throw new Error('Invalid agent template: missing container named noobaa-agent');
-    }
-
-    // Find/Create the agent conf env variable definition;
-    const agent_conf_var = agent_container.env
-        .find(env => env.name === 'AGENT_CONFIG');
-    if (!agent_conf_var) {
-        throw new Error('Invalid agent template: missing env variable definition named AGENT_CONFIG');
-    }
-
-    // Find the volume claim template.
-    const volume_claim_template = statefulset.spec.volumeClaimTemplates
-        .find(vct => vct.metadata.name === 'noobaastorage');
-    if (!volume_claim_template) {
-        throw new Error('Invalid agent template: missing volume claim template named noobaastorage');
-    }
+    const yaml_file = (await fs.readFileAsync(yaml_path)).toString();
```

Comment:
We moved to `fs.promises.readFile` instead, this fs.readFileAsync is deprecated/